esphome/__main__.py

```diff
@@ -346,6 +346,7 @@ def upload_program(config, args, host):
         not is_ip_address(CORE.address)
         and (get_port_type(host) == "MQTT" or config[CONF_MDNS][CONF_DISABLED])
         and CONF_MQTT in config
+        and not args.device
```

Comment:
`--device MQTT` is a valid cli option though...

Reply:
My apologies! MQTT isn't documented as a valid parameter for --device in the --help output and I missed that detail in my first read of the code.

The current behavior that I'm trying to fix attempts to do MQTT device discovery when specifying `--device IPADDRESS` if `mqtt:` is specified in the yaml configuration file. Historical behavior in this case was to use the OTA update mechanism with no MQTT involvement and that's what I'm trying to restore. 

After re-reading the code and testing the change, I see that changing line 347 from
`and (get_port_type(host) == "MQTT" or config[CONF_MDNS][CONF_DISABLED])` to 
`and (get_port_type(host) == "MQTT" and config[CONF_MDNS][CONF_DISABLED])` 
preserves the current MQTT device discovery feature (but only when both option `--device MQTT` and yaml configuration `mqtt:` are used) and restores the historical behavior when using option `--device IPADDRESS`. 

Does this seem reasonable? I'm new to contributing. If reasonable should I attempt to fix this pull request to make the change or should I close it and start anew? Thanks!

Reply:
You can probably just change it to this
```suggestion
        and (not args.device or args.device == "MQTT")
```